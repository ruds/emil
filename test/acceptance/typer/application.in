val tail = fn nil => nil | (::) (_, l) => l;
(tail [1, 2, 3], tail [4, 5, 6]);
(tail [1, 2, 3], tail ["foo", "bar", "baz"]);
val both = fn f => fn (a, b) => (f a, f b);
both tail ([1, 2, 3], [4, 5, 6]);
(both tail) ([1, 2, 3], [4, 5, 6]);
both tail ([1, 2, 3], ["foo", "bar", "baz"]);
