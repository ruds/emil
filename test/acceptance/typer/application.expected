@0001:
val tail : 'a list -> 'a list
@0002:
val it : (bigint list * bigint list)
@0003:
val it : (bigint list * string list)
@0004:
val both : ('a -> 'b) -> ('a * 'a) -> ('b * 'b)
@0005:
val it : (bigint list * bigint list)
@0006:
val it : (bigint list * bigint list)
@0007: ERROR
@0008:
